packages/client/src/util/datetimeConverters.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const isoRegex =
+  /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d+)?(?:Z|[+-]\d{2}:\d{2})?$/;
+
+/**
+ * Extracts the date from a ISO 8601 formatted date time string. Throws if the input is not in the correct format.
+ * @param dateTime An ISO 8601 formatted date time string
+ * @returns The date part of the input string
+ */
+export const extractDate = (dateTime: string) => {
+  if (!isoRegex.test(dateTime)) {
+    throw new Error(
+      `Invalid date format. Expected ISO 8601 format, but received: ${dateTime}`,
+    );
+  }
+
+  return dateTime.split(
```

Comment:
minor nit: Maybe could have another helper function for the string splitting logic since you do that in all 3 functions